### PR TITLE
feat: import constructors to import static methods

### LINKED ISSUE #134: Imported or wrapped constructs should use `from`
All functionless resources which can wrap existing CDK constructs should use a `from` import pattern rather than accept the construct in the constructor.

Example - EventBus (already follows this pattern):

```ts
functionless.EventBus.fromBus(new aws_events.EventBus(stack, name, props));
new functionless.EventBus(stack, "eventBus", props);
``` 

Example of a invalid pattern:

```ts
new functionless.Table(new aws_dynamo.Table(stack, "table"))
```

Should be 

```ts
functionless.Table.fromTable(new aws_dynamo.Table(stack, "table", props));
new Table(stack, "table", props);
```

* [x] Event Bridge
* [ ] Lambda - Part of #103 
* [ ] Dynamo (table)
* [ ] Step Functions
* [x] AppSync - Does not wrap a construct.

### PR DESCRIPTION
**Pull request by thantos**
Closes #134 

* `Table`
   * Add `fromTable`
   * Removed import constructor
   * Added creation construction `new Table(scope, id, props)`
* Function
   * Removed the deprecated import constructor
* Step Functions (`StepFunction` and `ExpressStepFunction`)
   * are no longer `Resource`s or `IStateMachine`s. They create or import a `StateMachine` and expose it via `.resource`.
   * add `fromStateMachine` to both

BREAKING_CHANGE: Removed `new Function(aws_lambda.Function)`. Use `Function.fromFunction`.
BREAKING_CHANGE: Removed `new Table(aws_dynamo.Table)`. Use `Table.fromTable` or `new Table(scope, id, {})`.
BREAKING_CHANGE: `StepFunction` and `ExpressStepFunction` are no longer CDK `Resources`. Use `StepFunction.resource` to get the `resource` generated by the constructor or imported using the new `StepFunction.fromStateMachine` or `ExpressStepFunction.fromStateMachine`.

### CODE REVIEW

File: test/__snapshots__/appsync.test.ts.snap [hunk 1]
```diff
@@ -7,72 +7,72 @@ Array [
...
+\\"version\\": \\"2018-05-29\\",
+\\"method\\": \\"POST\\",
+\\"resourcePath\\": \\"/\\",
+\\"params\\": {
+  \\"headers\\": {
+    \\"content-type\\": \\"application/x-amz-json-1.0\\",
+    \\"x-amz-target\\": \\"AWSStepFunctions.DescribeExecution\\"
+  },
+  \\"body\\": {
+    \\"executionArn\\": $util.toJson('exec1')
```
Comment (sam-goodwin) [inline test/__snapshots__/appsync.test.ts.snap:19]: Why did this change

Comment (thantos) [reply test/__snapshots__/appsync.test.ts.snap:19]: Looks like it just tabbed backwards
image

Ahh, found it : https://github.com/functionless/functionless/pull/249/files#diff-961ed18b565d502c0f3e3673413b522c59ba0121e9bec07aca2807cf73309addL1396-R1116

File: website/docs/error-codes.md [hunk 2]
```diff
@@ -39,3 +39,63 @@ During CDK synth a function was encountered which was not compiled by the Functi
 This suggests that the plugin was not correctly configured for this project.
 
 Ensure you follow the instructions at https://functionless.org/docs/getting-started.
+
+___
+
+### Function closure serialization was not allowed to complete
```
[the remark this answers is not in the document]
Comment (thantos) [reply website/docs/error-codes.md:45]: The large problem is that this file should have been removed... probably forgot to actually add it to the git ignore. 

Mentioned this elsewhere... doesn't make sense to track this generated file.

[the remark this answers is not in the document]
Comment (sam-goodwin) [reply website/docs/error-codes.md:45]: True for this file but I'm also seeing other changes

File: src/table.ts [hunk 3]
```diff
@@ -320,6 +381,82 @@ export class Table<
...
+  Kind extends string,
+  F extends AnyFunction
+> = IntegrationCall<`Table.${Kind}`, F>;
+
+/**
+ * Wraps an {@link aws_dynamodb.Table} with a type-safe interface that can be
+ * called from within an {@link AppsyncResolver}.
+ *
+ * Its interface, e.g. `getItem`, `putItem`, is in 1:1 correspondence with the
+ * AWS Appsync Resolver API https://docs.aws.amazon.com/appsync/latest/devguide/resolver-mapping-template-reference-dynamodb.html
```
[the remark this answers is not in the document]
Comment (thantos) [reply src/table.ts:394]: Its actually accurate right now, `table.*` can only be used in AppSync. Update this with #132 ?

[the remark this answers is not in the document]
Comment (sam-goodwin) [reply src/table.ts:394]: Right but it can be used with $AWS.DynamoDB

**Review by sam-goodwin — APPROVED** [review]

**Pull request merged**